src/js/libcs/OpImageDrawing.js

```diff
@@ -385,3 +385,169 @@ evaluator.cameravideo$0 = function() {
+    }
+    return res;
+}
+
+/**
+ * imagergba(‹image›,x,y) implements imagergb(‹imagename›,x,y) from Cinderella, i.e.
+ * returns a 4 component vector ranging from (0-255, 0-255, 0-255, 0-1)
+ */
+evaluator.imagergba$3 = function(args, modifs) {
+    var img = imageFromValue(evaluateAndVal(args[0]));
+    var x = evaluateAndVal(args[1]);
+    var y = evaluateAndVal(args[2]);
+
+    if (!img || x.ctype !== 'number' || y.ctype !== 'number') return nada;
+
+    x = Math.round(x.value.real);
+    y = Math.round(y.value.real);
+    if (isNaN(x) || isNaN(y)) return nada;
+
+    var rgba = readPixelsIndirection(img, x, y, 1, 1);
+    return List.realVector([rgba[0] * 255, rgba[1] * 255, rgba[2] * 255, rgba[3]]);
+};
+
+evaluator.imagergb$3 = evaluator.imagergba$3; //According to reference
+
+/**
+ * imagergba(<point1>, <point2>, ‹image›, <point3>) returns the color at the coordinate
+ * <point3> assuming that the left/right lower corner is <point1>/<point2> resp.
+ */
+evaluator.imagergba$4 = function(args, modifs) {
+    var img = imageFromValue(evaluateAndVal(args[2]));
+
+    var interpolate = true; //default values
+    var repeat = false;
+
+    function handleModifs() {
+        var erg;
+        if (modifs.interpolate !== undefined) {
+            erg = evaluate(modifs.interpolate);
+            if (erg.ctype === 'boolean') {
+                interpolate = (erg.value);
+            }
+        }
+
+        if (modifs.repeat !== undefined) {
+            erg = evaluate(modifs.repeat);
+            if (erg.ctype === 'boolean') {
+                repeat = (erg.value);
+            }
+        }
+    }
+    handleModifs();
+
+    if (!img) return nada;
+
+    var w = img.width;
+    var h = img.height;
+
+    var w0 = evaluateAndHomog(args[0]);
+    var w1 = evaluateAndHomog(args[1]);
+    var v0 = evaluateAndHomog(List.realVector([0, h, 1]));
+    var v1 = evaluateAndHomog(List.realVector([w, h, 1]));
+
+    if (w0 === nada || w1 === nada || p === nada) return nada;
+
+    //create an orientation-reversing similarity transformation that maps w0->v0, w1->v1
+    var ii = List.ii;
+    var jj = List.jj;
+
+    var m1 = eval_helper.basismap(v0, v1, ii, jj); //interchange I and J,
+    var m2 = eval_helper.basismap(w0, w1, jj, ii); //see Thm. 18.4 of Perspectives on Projective Geometry
+    var p = evaluateAndHomog(args[3]);
+    var coord = eval_helper.extractPoint(General.mult(m1, General.mult(List.adjoint3(m2), p)));
+
+    if (!coord.ok) return nada;
+
+    if (interpolate) {
+        coord.x -= 0.5; //center of pixels are in the middle of them.
+        coord.y -= 0.5; //Now pixel-centers have wlog integral coordinates
+    }
+
+    if (repeat) {
+        coord.x = (coord.x % w + w) % w;
+        coord.y = (coord.y % h + h) % h;
+    }
+
+    var rgba = [0, 0, 0, 0];
+
+    if (interpolate) {
+        var i, j;
+
+        var xf = (coord.x % 1 + 1) % 1; //fractional parts
+        var yf = (coord.y % 1 + 1) % 1;
```

Comment:
I find this hard to read. I understand that the `(… + 1) % 1` is designed to take care of situations where the initial remainder might have been negative. But since you will be using `Math.floor(coord.x)` in many places, why not make this more explicit:

```js
var xi = Math.floor(coord.x);
var xf = x - xi;
```

That way you can shorten the subsequent expressions and make this assignment here both easier to understand and faster to execute.